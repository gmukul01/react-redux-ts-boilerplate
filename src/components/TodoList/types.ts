import { Todos } from '@store/todos';

export interface Props {
    todos: Todos;
}
