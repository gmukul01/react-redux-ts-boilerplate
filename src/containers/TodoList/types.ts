import { Todos } from '@store/todos';

export interface StateProps {
    todos: Todos;
}

// We can merge all props here StateProps & DispatchProps & OwnProps and use this in the component
export type Props = StateProps;
