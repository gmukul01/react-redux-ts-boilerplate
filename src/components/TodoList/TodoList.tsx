import List from '@components/List';
import Text from '@components/Text';
import React, { useEffect } from 'react';
import { Props } from './types';

const TodoList: React.FC<Props> = ({ todos }) => (
    <List>
        {todos.map(todo => (
            <Text key={todo.id}>{todo.text}</Text>
        ))}
    </List>
);

export default TodoList;
