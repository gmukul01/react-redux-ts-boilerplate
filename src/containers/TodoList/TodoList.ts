import { TodoList } from '@components';
import { AppState } from '@store';
import { connect } from 'react-redux';
import { StateProps } from './types';

const mapStateToProps = ({ todos }: AppState): StateProps => ({
    todos
});

export default connect<StateProps>(
    mapStateToProps,
    null
)(TodoList);
