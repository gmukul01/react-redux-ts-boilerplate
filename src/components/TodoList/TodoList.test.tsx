import 'jest-styled-components';
import React from 'react';
import { render } from 'react-testing-library';

import TodoList from './TodoList';

describe('TodoList component', () => {
    it('should render with given todos', () => {
        const todos = [
                {
                    id: 1,
                    text: 'Random',
                    completed: false
                }
            ],
            { container } = render(<TodoList todos={todos} />);
        expect(container).toMatchSnapshot();
    });
});
