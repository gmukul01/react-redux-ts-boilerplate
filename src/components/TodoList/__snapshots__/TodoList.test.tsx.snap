// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList component should render with given todos 1`] = `
.c0 {
  padding: 0;
  margin: 0;
}

.c0 li * {
  display: block;
  margin: 4px 0px;
}

.c0 li:last-child * {
  margin-bottom: 8px;
}

.c0 li:first-child * {
  margin-top: 8px;
}

.c1 {
  font-size: 14px;
  font-weight: 400;
  color: inherit;
}

<div>
  <ul
    class="c0"
  >
    <li>
      <span
        class="c1"
      >
        Random
      </span>
    </li>
  </ul>
</div>
`;
